templates.js

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable no-console */
+const mongoose = require('mongoose');
+
+const fs = require('fs');
+const path = require('path');
+const util = require('util');
+const { minify } = require('html-minifier');
+const yamlConfig = require('node-yaml-config');
+const NotificationTemplate = require('./src/models/notificationTemplate');
+
+require('./config');
+
+const readFile = util.promisify(fs.readFile);
+
+const SUBJECTS_PATH = './src/subjects';
+const TEMPLATES_PATH = './src/templates';
+
+
+const envConfig = yamlConfig.load('./env.yml');
+const mongoDB = envConfig.MONGODB_URI;
+
+mongoose.connect(mongoDB);
+mongoose.Promise = global.Promise;
+const db = mongoose.connection;
+
+
+db.on('error', console.error.bind(console, 'MongoDB connection error:'));
+db.once('open', async () => {
+  console.log('Connected to MongoDB database');
+
```

Comment:
We should load the data into memory so we can do the following

- loop over the array in memory

  - if it has both a subject and template

    - if it exists in the db, update it.
    - if it doesn't exist in the db, create it. 

  - if it missing a subject or template, it is most likely an error, we should not update the record in the db. 

- delete templates that are in the db but not in the in-memory collection.